controllers/storagecluster/placement.go

```diff
@@ -130,3 +130,13 @@ func setTopologyForAffinity(placement *rookv1.Placement, selectorValue string, t
 	}
 	appendNodeRequirements(placement, nodeZoneSelector)
 }
+
+// setMultipleTSC assigns hard topology constraints at failure domain level
+// and soft topology constraints within falure domain (across host).
+func setMultipleTSC(placement *rookv1.Placement, topologyKey string) {
```

Comment:
This function name is too generic for what it actually does. Rename this to something like `addStrictFailureDomainTSC()`. Also, move this to cephcluster.go